packages/ListBox/src/components/Trigger/Trigger.js

```diff
@@ -1,100 +1,84 @@
   sanitizeActionTypes,
   getSelectedOptionSingle,
   getSelectedOptionsMulti,
 } from "../../helpers/invokeOnChange";
-
 import * as sc from "./Trigger.styles";
 import { getDOMAttributesForListBoxButton } from "../../helpers/DOMAttributes";
 
-const propTypes = {
-  /** Custom clear icon */
-  clearIcon: PropTypes.node,
-
-  /** Body content of the trigger. */
-  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
-
-  /** If true it adds a clear button */
-  hasClearButton: PropTypes.bool,
-
-  /** Has implicit "All items selected" value when no item is selected */
-  hasImplicitAll: PropTypes.bool,
-
-  /** Override the label with a custom one. */
-  label: PropTypes.string,
-
-  /** Callback to be executed when the clear button is clicked or activated by keyboard. */
-  onClickClear: PropTypes.func,
-
-  /** Callback to be executed when the accept button is clicked or activated by keyboard. */
-  onClickFooterAccept: PropTypes.func,
-
-  /** Sets a placeholder for the trigger */
-  placeholder: PropTypes.string,
-
-  /** If true the trigger will be hidden */
-  isHidden: PropTypes.bool,
-};
-
-const defaultProps = {
-  clearIcon: null,
-  children: <React.Fragment />,
-  hasClearButton: true,
-  hasImplicitAll: false,
-  label: null,
-  onClickClear: null,
-  onClickFooterAccept: null,
-  placeholder: "Select...",
-  isHidden: false,
-};
-
 export default function Trigger(props) {
+  const I18n = useI18n();
   const [state, dispatch] = useListBox();
   const onChangeContext = React.useContext(OnChangeContext);
+  const providedProps = React.useContext(PropsContext);
+
   const {
+    children,
     clearIcon,
-    placeholder,
     hasClearButton,
     hasImplicitAll,
-    label,
-    onClickFooterAccept,
-    children,
     isHidden,
+    label,
     onClickClear,
-    // eslint-disable-next-line react/prop-types
-    onFooterClickAccept,
+    onClickFooterAccept,
+    morePropsForTrigger,
     ...moreProps
   } = props;
+
+  const { hasError, idListBox, isDisabled, isInline, isReadOnly, size } = providedProps;
+
   const {
-    isDisabled,
-    formElementLabelDescribedBy,
-    refTriggerContainer,
-    refTrigger,
+    activeOption,
+    hasFooter,
+    isOpen,
     isMulti,
-    idListBox,
+    onChange,
+    options,
     refLabel,
-    size,
+    refTrigger,
+    refTriggerContainer,
+    selectedOptions,
   } = state;
 
-  const [triggerButtonId] = React.useState(() => `list-box-trigger_${uuidv4()}`);
-  const I18n = useI18n();
+  const placeholder = props.placeholder || providedProps.placeholder;
+
+  const idFormLabel = React.useRef();
+  React.useEffect(() => {
+    const $label = refLabel && refLabel.current;
+    if (!$label) return;
+
+    const possibleId = $label.getAttribute("id");
+    if (possibleId) {
+      idFormLabel.current = possibleId;
+    } else {
+      const newId = `${idListBox}__form-label`;
+      $label.setAttribute("id", newId);
+      idFormLabel.current = newId;
+    }
```

Comment:
This is new.  The `<FormElement.Label>` element needs an `id` to be used later with `aria-labelledby`.

Reply:
`if(refLabel?.current instanceof Element)` wouldn't you want to test is actually a dom element? or we can assume is a dom element? 

wouldn't be better if you always write a new id?, you wouldn't have to worry about it if has it or not, just always overwrite it? 

Also, super side option, I wonder if you can take control of the labe like the following and then tweak it at your pleasure in a more reactify way.

`ReactDOM.render(<Lable id="">your stuff</Label>, refLabel.current)`

Reply:
> wouldn't you want to test is actually a dom element?

Good idea, will do.

> wouldn't be better if you always write a new id? 
> I wonder if you can take control of the labe like the following and then tweak it at your pleasure in a more reactify way.

These seem too obtrusive to me for a component to be imposing such side effects on a sibling or parent element/component.  I think it's the wrong direction of control.  I'm a little hesitant about even writing the `id` attribute, which is why I check first to see if one exists rather than just overwriting – if there's no change, just addition, it seems less disruptive, but it's still the wrong direction of control.  Just the best solution I could think of without requiring a change to, and stronger coupling with, the `FormElement`.